File: src/string/getSumOfDigits/test_get_sum_of_digits.py
from get_sum_of_digits import *
import pytest


@pytest.mark.parametrize('input, expected', [
    ('2514', 12),
    ('12', 3),
    ('12345', 15),
    ('7777777777', 70)
])
def test_get_sum_of_digits(input, expected):
    assert get_sum_of_digits(input) == expected

File: src/string/getSumOfDigits/TestGetSumOfDigits.java
package string.getSumOfDigits;

import static org.junit.Assert.assertEquals;
import static string.getSumOfDigits.GetSumOfDigits.getSumOfDigits;

import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.CsvSource;

public class TestGetSumOfDigits {

    @ParameterizedTest
    @CsvSource(value = {
            "2514, 12",
            "12, 3",
            "12345, 15",
            "7777777777, 70",
    })
    void testGetSumOfDigits(String input, int expected) {
        assertEquals(expected, getSumOfDigits(input));
    }
}